lib/pavilion/series.py

```diff
@@ -36,12 +49,236 @@ def test_obj_from_id(pav_cfg, test_ids):
+
+            self.check_and_update()
+            temp_waiting = copy.deepcopy(self.not_started)
+            for test_name in temp_waiting:
+                ready = all(wait in self.finished for wait in self.test_info[
+                    test_name]['prev'])
+                if ready:
+                    if self.series_section[test_name]['depends_pass'] == 'True':
+                        if self.all_tests_passed(
+                                self.test_info[test_name]['prev']):
+                            self.run_test(test_name)
+                            self.not_started.remove(test_name)
+                        else:
+                            temp_resolver = resolver.TestConfigResolver(
+                                self.pav_cfg)
+                            raw_configs = temp_resolver.load_raw_configs(
+                                [test_name], [],
+                                self.test_info[test_name]['modes']
+                            )
+                            for config in raw_configs:
+                                # need to delete these so they won't get
+                                # evaluated
+                                del config['only_if']
+                                del config['not_if']
+                                skipped_test = TestRun(self.pav_cfg, config)
+                                skipped_test.status.set(
+                                    STATES.SKIPPED,
+                                    "Skipping. Previous test did not PASS.")
+                                skipped_test.set_run_complete()
+                                self.series_obj.add_tests([skipped_test])
+                            self.not_started.remove(test_name)
+                            self.finished.append(test_name)
+                    else:
+                        self.run_test(test_name)
+                        self.not_started.remove(test_name)
+
+            time.sleep(1)
+
+    def run_test(self, test_name):
+        # basically copy what the run command is doing here
+        mb_tracker = MultiBuildTracker()
+
+        sys_vars = system_variables.get_vars(True)
+
+        run_cmd = commands.get_command('run')
+
+        # run.RunCommand._get_tests function
+        try:
+            new_conditions = {
+                'only_if': self.test_info[test_name]['only_if'],
+                'not_if': self.test_info[test_name]['not_if']
+            }
+
+            # resolve configs
+            # pylint: disable=W0212
+            configs_by_sched = run_cmd._get_test_configs(
+                pav_cfg=self.pav_cfg,
+                host=None,
+                test_files=[],
+                tests=[test_name],
+                modes=self.test_info[test_name]['modes'],
+                overrides=None,
+                sys_vars=sys_vars,
+                conditions=new_conditions
+            )
+
+            # configs -> test
+            # pylint: disable=W0212
```

Comment:
Convert this to a public function.

Reply:
Are you referring to `def run_test(self, test_name)`?